src/http/conn.rs

```diff
@@ -235,12 +237,17 @@ where I: AsyncRead + AsyncWrite,
 
         let wants_keep_alive = head.should_keep_alive();
         self.state.keep_alive &= wants_keep_alive;
-        let mut buf = Vec::new();
+
+        // TODO: Can we reuse this butter?
+        // TODO: Figure out a better init capacity
+        let mut buf = BytesMut::with_capacity(0);
```

Comment:
I've toyed around with this a bit, actually. Using a `Vec` is actually faster here. `BytesMut` has some atomic checks inside each of its methods, and since we never share this buffer to other threads, we don't need it.

I have toyed with passing `self.io.write_buf_mut()` directly, instead, since it is likely an already-allocated vector, and would remove the copy from the intermediate vector into the write buf. I couldn't tell with my machine if it made any noticeable change, and was also worried about the `reserve`s inside `encode` making the buffer bigger than the max buf size, so, it hasn't changed. I'd welcome a change with measurements showing an improvements!